src/ahp/indexer.rs

```diff
@@ -176,22 +188,20 @@ impl<F: PrimeField> AHPForR1CS<F> {
             .ok_or(SynthesisError::PolynomialDegreeTooLarge)?;
         let domain_k = GeneralEvaluationDomain::new(num_non_zero)
             .ok_or(SynthesisError::PolynomialDegreeTooLarge)?;
-        let x_domain = GeneralEvaluationDomain::<F>::new(num_formatted_input_variables)
+        let x_domain = GeneralEvaluationDomain::new(num_formatted_input_variables)
             .ok_or(SynthesisError::PolynomialDegreeTooLarge)?;
-        let b_domain = GeneralEvaluationDomain::<F>::new(3 * domain_k.size() - 3)
-            .ok_or(SynthesisError::PolynomialDegreeTooLarge)?;
-
-        let a_arithmetization_time = start_timer!(|| "Arithmetizing A");
-        let a_star_arith = arithmetize_matrix("a", &mut a, domain_k, domain_h, x_domain, b_domain);
-        end_timer!(a_arithmetization_time);
 
-        let b_arithmetization_time = start_timer!(|| "Arithmetizing B");
-        let b_star_arith = arithmetize_matrix("b", &mut b, domain_k, domain_h, x_domain, b_domain);
-        end_timer!(b_arithmetization_time);
-
-        let c_arithmetization_time = start_timer!(|| "Arithmetizing C");
-        let c_star_arith = arithmetize_matrix("c", &mut c, domain_k, domain_h, x_domain, b_domain);
-        end_timer!(c_arithmetization_time);
+        let joint_arithmetization_time = start_timer!(|| "Arithmetizing A");
```

Comment:
"Arithmetizing A" -> "Arithmetizing the joint matrix"?

Reply:
```suggestion
        let joint_arithmetization_time = start_timer!(|| "Arithmetizing all matrices");
```